Client.Linq/Internal/QueryExpressionTreeVisitor.cs

```diff
@@ -187,6 +187,88 @@ protected override Expression VisitMethodCall(MethodCallExpression expression)
+                    _expressionParts.Add(new StringFunction("containsStr", part, part2, null));
+                    return expression;
+                }
+                else if (expression.Method.Name.Equals("StartsWith"))
+                {
+                    _context.QueryAggregator.AddImport("strings");
+                    var partsCount = _expressionParts.Count;
+                    Visit(expression.Object);
+                    var part = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    partsCount = _expressionParts.Count;
+                    Visit(expression.Arguments[0]);
+                    var part2 = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    _expressionParts.Add(new StringFunction("hasPrefix", part, part2, null));
+                    return expression;
+                }
+                else if (expression.Method.Name.Equals("EndsWith"))
+                {
+                    _context.QueryAggregator.AddImport("strings");
+                    var partsCount = _expressionParts.Count;
+                    Visit(expression.Object);
+                    var part = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    partsCount = _expressionParts.Count;
+                    Visit(expression.Arguments[0]);
+                    var part2 = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    _expressionParts.Add(new StringFunction("hasSuffix", part, part2, null));
+                    return expression;
+                }
+                else if (expression.Method.Name.Equals("Replace"))
+                {
+                    _context.QueryAggregator.AddImport("strings");
+                    var partsCount = _expressionParts.Count;
+                    Visit(expression.Object);
+                    var part = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    partsCount = _expressionParts.Count;
+                    Visit(expression.Arguments[0]);
+                    var part2 = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    partsCount = _expressionParts.Count;
+                    Visit(expression.Arguments[1]);
+                    var part3 = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                    _expressionParts.Add(new StringFunction("replaceAll", part, part2, part3));
+                    return expression;
+                }
+            }
+
+            if (expression.Method.Name.Equals("ToString"))
+            {
+                var partsCount = _expressionParts.Count;
+                Visit(expression.Object);
+                var part = GetAndRemoveExpressionParts(partsCount, _expressionParts.Count);
+                _expressionParts.Add(new StringFunction("string", part, null, null));
```

Comment:
Please create a `ToString` class which will represent the `string` expression, instead of using the generic `StringFunction` class. This approach will enhance future maintenance and also facilitate contributions from other users.